src/main/java/net/sf/jabref/shared/prefs/SharedDatabasePreferences.java

```diff
@@ -0,0 +1,90 @@
+package net.sf.jabref.shared.prefs;
+
+import java.util.Optional;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.gui.shared.OpenSharedDatabaseDialog;
+
+/**
+ * Stores and reads persistent data for {@link OpenSharedDatabaseDialog}.
+ */
+public class SharedDatabasePreferences {
+
+    private static final String SHARED_DATABASE_TYPE = "sharedDatabaseType";
+    private static final String SHARED_DATABASE_HOST = "sharedDatabaseHost";
+    private static final String SHARED_DATABASE_PORT = "sharedDatabasePort";
+    private static final String SHARED_DATABASE_NAME = "sharedDatabaseName";
+    private static final String SHARED_DATABASE_USER = "sharedDatabaseUser";
+    private static final String SHARED_DATABASE_PASSWORD = "sharedDatabasePassword";
+    private static final String SHARED_DATABASE_REMEMBER_PASSWORD = "sharedDatabaseRememberPassword";
+
+    // This {@link Preferences} is used only for things which should not appear in real JabRefPreferences due to security reasons.
+    private final Preferences internalPrefs = Preferences.userNodeForPackage(OpenSharedDatabaseDialog.class);
```

Comment:
Is it possible to use another node for storing? The whole preference tree gets exported. Therefore, this should be located somewhere else @matthiasgeiger